### PR TITLE
e2e: serial lane cleanup and fixes

### PR DESCRIPTION
**Pull request by fromanirh**
In the latest internal CI cluster run we have a string of failures:
```
Summarizing 9 Failures:

[Fail] [serial][disruptive][scheduler] numaresources workload placement considering taints when cluster has two feasible nodes with taint but only one has the requested resources on a single NUMA zone [BeforeEach] [test_id:47594][tier1] should make a pod with a toleration land on a node with enough resources on a specific NUMA zone 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement_taint.go:111

[Fail] [serial][disruptive][scheduler] numaresources workload placement Requesting resources that are greater than allocatable at numa level [It] [test_id:47613][tier3] should not schedule a pod requesting resources that are not allocatable at numa level 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/non_regression.go:369

[Fail] [serial][disruptive][scheduler] numaresources workload placement cluster has at least one suitable node [It] [test_id:47591][tier1] should modify workload post scheduling while keeping the resource requests available 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement.go:219

[Fail] [serial][disruptive][scheduler] numaresources workload placement cluster has at least one suitable node [It] [test_id:47628][tier2] should schedule a workload (with TAS) and then keep a subsequent one (with default scheduler) pending  
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement.go:456

[Fail] [serial][disruptive][scheduler] numaresources workload placement cluster has at least one suitable node with Topology Manager single numa policy (both container and pod scope acceptable) [It] [test_id:48746][tier2] should modify workload post scheduling while keeping the resource requests available across all NUMA node 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement.go:745

[Fail] [serial][disruptive][scheduler] numaresources workload placement cluster with at least two available nodes [It] [test_id:47627] should be able to schedule many replicas with performance time equals to the default scheduler 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement.go:967

[Fail] [serial][disruptive][scheduler] numaresources workload placement considering TM policy [placement] cluster with multiple worker nodes suitable [It] [tier1][testtype11][tmscope:container] should make a pod with one init cnt and three gu cnt land on a node with enough resources, containers should be spread on a different zone 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement_tmpol.go:476

[Fail] [serial][disruptive][scheduler] numaresources workload placement considering TM policy [placement] cluster with multiple worker nodes suitable [It] [tier1][testtype29][tmscope:container] should make a pod with 3 gu cnt and 3 init cnt land on a node with enough resources, when sum of init and app cnt resources are more than node resources 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_placement_tmpol.go:476

[Fail] [serial][disruptive][scheduler] numaresources workload unschedulable with zero suitable nodes [It] [test_id:47615][tier2][unsched] a deployment with multiple guaranteed pods resources that doesn't fit at the NUMA level 
/home/fromani/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests/workload_unschedulable.go:688
```

This PR addresses them

--

UPDATE 20220909 with the fixes in this PR all the failures except `testtype11` and `testtype29` are now fixed. We gathered quite some changes already, so I think it's best to move forward and to address these failures on a separate PR

### CODE REVIEW

File: test/e2e/serial/tests/workload_placement.go [hunk 1]
```diff
@@ -114,7 +114,12 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
 				corev1.ResourceMemory: resource.MustParse("8G"),
 			}
 			By("padding the nodes before test start")
-			err := padder.Nodes(numOfNodeToBePadded).UntilAvailableIsResourceList(rl).Pad(timeout, e2epadder.PaddingOptions{})
```
Comment (shajmakh) [inline test/e2e/serial/tests/workload_placement.go:117]: I get the idea of the fix, yet I believe tests that requires nrtcandidates that are >=2 should have been skipped and not reach this code, in this specific case the test should be skipped here:
https://github.com/openshift-kni/numaresources-operator/blob/3abdd3cde6213800f43ceaee6d4603b1a1501e9c/test/e2e/serial/tests/workload_placement.go#L106

Comment (shajmakh) [reply test/e2e/serial/tests/workload_placement.go:117]: ah this points to the nrts with the policies. maybe applying similar solution here to what you've done in the second commit

Comment (fromanirh) [reply test/e2e/serial/tests/workload_placement.go:117]: true. The problem is that the padder pkg will try to do -earlier than that point- a padding of the candidate nodes until the free resources are what is declared in `rl` above. Thus 3 cpus, 8G of RAM. In some (arguably debatable) CI setup we can have virtual worker nodes mixed with physical worker nodes, and the virtual nodes are usually way weaker than physical worker.

Arguably, albeit that's a pretty weak stance, we should do check the available resources on nodes before we attempt to pad them. This fix only partially address this issue, and incidentally.

**Comment by shajmakh** [comment]
Thanks!
/lgtm

**Comment by fromanirh** [comment]
too early, these fixes aren't enough to fix. I get the aforementioned failures _even with these fix applied_. More work is needed here.

**Comment by fromanirh** [comment]
with the fixes in this PR all the failures except testtype11 and testtype29 are now fixed. We gathered quite some changes already, so I think it's best to move forward and to address these failures on a separate PR.

File: test/e2e/serial/tests/workload_placement.go (hunk 1 above)
Comment (fromanirh) [reply test/e2e/serial/tests/workload_placement.go:117]: I managed to improve this in the followup commits, but a overhaul is probably still in order (material for future PRs)

**Pull request merged**